### PR TITLE
utils_env: set protocol to 0 for pickle.dump

### PR DESCRIPTION
**Pull request by hereischen**
Python 2's default protocol is 0, whereas Python3.x is 3 or 4
depending on the x.
For VirNet's attribute, if its value is empty string or None,
it will be preserved when protocol is 0 or 1, the rest protocol are not.
So use protocol 0 on both Python3 and 2 to make it compatiable.

Revert "env_process: invoke vm.destroy when vm is alive"
This reverts commit d363a0b.

Signed-off-by: Haotong Chen <hachen@redhat.com>

id: 1611920

### CODE REVIEW

**Comment by hereischen** [comment]
@pingl2016 it may not clean all resources up

**Comment by hereischen** [comment]
09:04:14 ERROR| Traceback (most recent call last):
09:04:14 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_plugins_vt-63.0-py3.6.egg/avocado_vt/test.py", line 297, in runTest
09:04:14 ERROR|     raise self.__status  # pylint: disable=E0702
09:04:14 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_plugins_vt-63.0-py3.6.egg/avocado_vt/test.py", line 251, in setUp
09:04:14 ERROR|     self._runTest()
09:04:14 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_plugins_vt-63.0-py3.6.egg/avocado_vt/test.py", line 423, in _runTest
09:04:14 ERROR|     run_func(self, params, env)
09:04:14 ERROR|   File "/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/qemu_disk_img_convert.py", line 76, in run
09:04:14 ERROR|     convert_test.start_vm(n_params)
09:04:14 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_plugins_vt-63.0-py3.6.egg/virttest/error_context.py", line 135, in new_fn
09:04:14 ERROR|     return fn(*args, **kwargs)
09:04:14 ERROR|   File "/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/qemu_disk_img.py", line 66, in start_vm
09:04:14 ERROR|     env_process.preprocess_vm(self.test, params, self.env, vm_name)
09:04:14 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_plugins_vt-63.0-py3.6.egg/virttest/env_process.py", line 185, in preprocess_vm
09:04:14 ERROR|     old_vm.destroy(gracefully=gracefully_kill)
09:04:14 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_plugins_vt-63.0-py3.6.egg/virttest/qemu_vm.py", line 3198, in destroy
09:04:14 ERROR|     self._cleanup(free_mac_addresses)
09:04:14 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_plugins_vt-63.0-py3.6.egg/virttest/qemu_vm.py", line 3125, in _cleanup
09:04:14 ERROR|     elif nic.ifname and nic.ifname not in utils_net.get_net_if():
09:04:14 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_plugins_vt-63.0-py3.6.egg/virttest/propcan.py", line 230, in __getattr__
09:04:14 ERROR|     return super(PropCanBase, self).__getattribute__(key)
09:04:14 ERROR| AttributeError: ifname

**Comment by hereischen** [comment]
RESULTS    : PASS 93 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 1 | INTERRUPT 0 | CANCEL

**Comment by luckyh** [comment]
@clebergnu @ldoktor @balamuruhans please help review this one, thanks!

File: virttest/env_process.py [hunk 1]
```diff
@@ -416,8 +415,7 @@ def postprocess_vm(test, params, env, name):
         kill_vm_timeout = float(params.get("kill_vm_timeout", 0))
         if kill_vm_timeout:
             utils_misc.wait_for(vm.is_dead, kill_vm_timeout, 0, 1)
-        if vm.is_alive():
-            vm.destroy(gracefully=params.get("kill_vm_gracefully") == "yes")
+        vm.destroy(gracefully=params.get("kill_vm_gracefully") == "yes")
```
[the remark this answers is not in the document]
Comment (hereischen) [reply virttest/env_process.py]: reason added, thanks

**Comment by balamuruhans** [comment]
Looks good to me, @pingl2016, request for your ACK.

**Comment by balamuruhans** [comment]
Thanks all, let's merge it.

**Pull request merged**